python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -463,11 +463,14 @@ def _keep_min_worker_of_node_type(
             node_type_counts(Dict[NodeType, int]): The non_terminated node
                 types counted so far.
         Returns:
-            bool: if workers of node_types can be terminated or not.
+            bool: True if the node should be kept. False otherwise.
         """
         tags = self.provider.node_tags(node_id)
         if TAG_RAY_USER_NODE_TYPE in tags:
             node_type = tags[TAG_RAY_USER_NODE_TYPE]
+            if node_type not in self.available_node_types:
+                # The user has deleted this node type - don't keep the node.
```

Comment:
“The node type is deleted from the cluster yaml...

Reply:
Yeah, much clearer.